File: Basic Exercises/Easy/7. Reverse Integer/7. Reverse Integer.java
class Solution {
    public int reverse(int x) {
        int ans=0;
        while(x!=0){
            int mod=x%10;
            if (ans > Integer.MAX_VALUE/10 || (ans == Integer.MAX_VALUE / 10 && mod > 7)) return 0;
            if (ans < Integer.MIN_VALUE/10 || (ans == Integer.MIN_VALUE / 10 && mod < -8)) return 0;
            ans=10*ans+mod;
            x/=10;
        }
        return ans;
    }
}

File: Basic Exercises/Easy/7. Reverse Integer/7. Reverse Integer.py
class Solution:
    def reverse(self, x: int) -> int:
        flag=-1 if x<0 else 1
        x*=flag
        ans=0
        while(x):
            if not -2**31<=ans<=2**31-1:
                return 0
            ans=10*ans+x%10
            x//=10
        ans*=flag
        return ans if -2**31<=ans<=2**31-1 else 0
